snmp/datadog_checks/snmp/snmp.py

```diff
@@ -281,16 +281,24 @@ def check(self, instance):
         Perform two series of SNMP requests, one for all that have MIB associated
         and should be looked up and one for those specified by oids.
         """
-        # Reset errors
-        self._error = self._severity = None
         if instance.get('network_address'):
-            for discovered in list(self._config.discovered_instances.values()):
-                # TODO: cleanup failing hosts
-                self._check_with_config(discovered)
+            for host, discovered in list(self._config.discovered_instances.items()):
+                if self._check_with_config(discovered):
```

Comment:
Since `_error` is an attribute of `self`. You don't have to move `self._error` inside `_check_with_config` and make it return.

So it could just be:

```python
self._check_with_config(discovered)
if self. _error:
    ...
```

Reply:
My bad, seems the errors need to be reset for each  `self._check_with_config`  call.